### PR TITLE
4201 add sponser's donations manually

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
Related to https://github.com/Giveth/giveth-dapps-v2/issues/4201

## Summary by CodeRabbit

- **New Features**
  - Introduced Project Donation Summary view for better insight into project donations.
  - Added functionality to relate current mini stream donations to QF rounds.

- **Bug Fixes**
  - Improved error handling and logging in project verification processes.
  - Corrected admin user ID handling across various test cases and repositories.

- **Refactor**
  - Replaced `admin` field with `adminUserId` for clarity and consistency.
  - Simplified rollback methods in migration scripts.

- **Chores**
  - Added new dependencies and test scripts in `package.json`.
  - Implemented caching mechanisms to optimize query performance.

### CODE REVIEW

**Comment by maryjaf** [comment]
How can I test this ? @mohammadranjbarz

**Comment by mohammadranjbarz** [comment]
> How can I test this ? @mohammadranjbarz

It's not merged yet, but after emerging this you should see the donations in the matching-pool project page

File: migration/1716549958362-add_donations_mannually_to_db.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+             true, true, true, '${createdAt}', ${givbackFactor}, ${powerRound}, ${projectRank}, ${bottomRankInRound}, ${tx.qfRoundId || null}, '${
+               tx.tokenAddress
+             }');
+                `);
+
+      await updateUserTotalDonated(user.id);
+      await updateUserTotalReceived(project.adminUser?.id);
+      await updateTotalDonationsOfProject(tx.projectId as number);
+    }
+
```
Comment (CarlosQ96) [inline migration/1716549958362-add_donations_mannually_to_db.ts:405]: I added the method updateProjectStatistics() for the project_donation_summary_view, maybe we have to call it here too. 

The rest looks good.

**Review by CarlosQ96 — CHANGES_REQUESTED** [review]
To update totals we have to call the method. AFter that, can merge

File: migration/1716549958362-add_donations_mannually_to_db.ts (hunk 1 above)
Comment (CarlosQ96) [reply migration/1716549958362-add_donations_mannually_to_db.ts:405]: Might need to rebase master into this branch

**Review by jainkrati — APPROVED** [review]

File: migration/1716549958362-add_donations_mannually_to_db.ts (hunk 1 above)
Comment (mohammadranjbarz) [reply migration/1716549958362-add_donations_mannually_to_db.ts:405]: Thanks @CarlosQ96

**Pull request merged**